sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2665,6 +2665,14 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val ANSI_FAIL_ON_ELEMENT_NOT_EXISTS = buildConf("spark.sql.ansi.failOnElementNotExists")
```

Comment:
no biggie but maybe `failOnElementNonExistence`? no strong opinion. LGTM as is 2.